### PR TITLE
Feat/live 3533 swap provider list

### PR DESCRIPTION
**Pull request by sarneijim**
### 📝 Description

Improving swap page for the user to easily compare options

### ❓ Context

- **Impacted projects**: `` 
  - Common
  -  LLD
- **Linked resource(s)**:  https://ledgerhq.atlassian.net/browse/LIVE-3533
  - [swap] As a user on desktop, I want to be able to see the provider list upfront  LIVE-3534 https://ledgerhq.atlassian.net/browse/LIVE-3534
  - [swap] As a user on desktop, I want to distinguish fix vs. floating rate LIVE-3535 https://ledgerhq.atlassian.net/browse/LIVE-3535 
  - [swap] As a user on desktop, I want to know when the market price expired and are refreshed LIVE-3536 https://ledgerhq.atlassian.net/browse/LIVE-3536
  -  [swap] As a user on desktop, I want to identify which option require an account to be created or not LIVE-3537 https://ledgerhq.atlassian.net/browse/LIVE-3537
  - [swap] As a user on desktop, I want to see a tooltip on top of each column LIVE-3538 https://ledgerhq.atlassian.net/browse/LIVE-3538

### ✅ Checklist

- [x] **Test coverage** 
- [ ] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

Screenshot 2022-09-08 at 18 59 40

### 🚀 Expectations to reach

Improving swap page for the user to easily compare options
_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: libs/ledger-live-common/src/exchange/swap/utils/index.ts [hunk 1]
```diff
@@ -172,6 +172,11 @@ export const shouldShowKYCBanner = ({
   return kycStatus !== KYC_STATUS.approved;
 };
 
+export const registrationIsRequired = (provider: string): boolean => {
```
Comment (chabroA) [inline libs/ledger-live-common/src/exchange/swap/utils/index.ts:175]:
```suggestion
export const isRegistrationRequired = (provider: string): boolean => {
```

For better naming convention, name Functions as verbs.
cf. https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0

**Review by chabroA — CHANGES_REQUESTED** [review]
Good job overall but the component layout regarding the new rates list needs to be updated. It is currently sub optimal and constantly re-rendered (every second), triggering some UI problems / glitches + overall poor performance

cf. following videos:

-----

https://user-images.githubusercontent.com/9203826/189660144-a2d31215-2bac-4d8c-a175-66da08e4ffe9.mov

https://user-images.githubusercontent.com/9203826/189660277-06b90650-f4be-49cf-ab6c-45ad12719ad0.mov


https://user-images.githubusercontent.com/9203826/189660374-e9b00dd2-d50b-416d-946b-354f5a7ed005.mov

In comparaison, here is how it behaves on the current prod version (only the countdown is re-rendered):

https://user-images.githubusercontent.com/9203826/189660720-4a63eca2-3149-4a62-b0b6-2f28a1508210.mov

This is mainly due to how you broke down the component hierarchy of this new provider / rates list. And also due to the fact that you handle the countdown at the root form component (`Swap2/Form/index.jsx`)

Here is an example of how it could be broken down:

You would only handle the countdown logic in the red component, hence only this component would be re-rendered every second, and not the whole form (it could be broken down even more with just the countdown being it's own atomic component, like it is done today).

Screenshot 2022-09-12 at 14 31 09

For the countdown UI, instead of using a new png image for the timer, you could use the already existing `AnimatedCountdown` component that is currently used in prod.

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.jsx [hunk 2]
```diff
@@ -156,6 +159,15 @@ const SwapForm = () => {
...
   const { setDrawer } = React.useContext(context);
 
+  useEffect(() => {
+    const refreshInterval = setInterval(() => {
+      swapTransaction?.swap?.refetchRates();
+    }, refreshTime);
+    return () => {
+      clearInterval(refreshInterval);
+    };
+  }, [swapTransaction.swap.from.amount]);
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.jsx:169]: Should this not also include anything from the swap (to, provider, etc)?

Comment (ggilchrist-ledger) [reply apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.jsx:169]: Ah I see, only the from can be changed, not the 'To', and the whole quote is eventually updated when provider is changed. The only thing is there are a few seconds when the timer for the previous quote still counts down when the provider has changed. 

https://user-images.githubusercontent.com/87659842/189901083-e0559990-0519-4679-ad10-b72a40c4b16e.mov

**Comment by ggilchrist-ledger** [comment]
@sarneijim I think the main 'Providers' heading should be 'Quotes', and 'Float' should be floating.
image

Also the info tag for 'Receive' is a bit off center: 
image

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.jsx (hunk 2 above)
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.jsx:170]: This is the bug that I was talking this morning. 
In the new version, I am restoring the data when the user change "From" or "To".

File: libs/ledger-live-common/src/exchange/swap/utils/index.ts (hunk 1 above)
Comment (sarneijim) [reply libs/ledger-live-common/src/exchange/swap/utils/index.ts]: Done

**Comment by sarneijim** [comment]
> Good job overall but the component layout regarding the new rates list needs to be updated. It is currently sub optimal and constantly re-rendered (every second), triggering some UI problems / glitches + overall poor performance
> 
> cf. following videos:
> 
>  Screen.Recording.2022-09-12.at.13.59.54.mov 
>  Screen.Recording.2022-09-12.at.14.27.12.mov 
>  Screen.Recording.2022-09-12.at.14.28.07.mov 
> In comparaison, here is how it behaves on the current prod version (only the countdown is re-rendered):
> 
>  Screen.Recording.2022-09-12.at.14.40.32.mov 
> This is mainly due to how you broke down the component hierarchy of this new provider / rates list. And also due to the fact that you handle the countdown at the root form component (`Swap2/Form/index.jsx`)
> 
> Here is an example of how it could be broken down:
> 
> You would only handle the countdown logic in the red component, hence only this component would be re-rendered every second, and not the whole form (it could be broken down even more with just the countdown being it's own atomic component, like it is done today).
> 
> Screenshot 2022-09-12 at 14 31 09
> 
> For the countdown UI, instead of using a new png image for the timer, you could use the already existing `AnimatedCountdown` component that is currently used in prod.

Fixed glitches problem, and we are using now the countdown component. I would ask Diego to check that he is agree with it.

**Comment by sarneijim** [comment]
> @sarneijim I think the main 'Providers' heading should be 'Quotes', and 'Float' should be floating. image
> 
> Also the info tag for 'Receive' is a bit off center: image

Changed

**Review by chabroA — APPROVED** [review]
Looks good overall. Major issues from previous review are fixed.

PS: I had to fix some dead code, typing and styling issues. It looks like your environment and / or tooling might not be properly setup

**Comment by ggilchrist-ledger** [comment]
It looks like the timer is still causing a warning in the dev tools. Not sure if its an issue or not. Can you check? @sarneijim @alex-ledger 

https://user-images.githubusercontent.com/87659842/190093116-6524d3b0-5670-4402-8bcd-8d445434ee17.mov

**Comment by alex-ledger** [comment]
Hi. Did you mentioned correct "alex"?😁 @ggilchrist-ledger

**Comment by ggilchrist-ledger** [comment]
Oops no I didn't! Sorry @alex-ledger

**Comment by ggilchrist-ledger** [comment]
Can you fix the wrapping of the text here so it occupies the full space (like the designs)? @sarneijim 
Screenshot 2022-09-14 at 10 23 43

File: apps/ledger-live-desktop/static/i18n/en/app.json [hunk 3]
```diff
@@ -427,7 +427,13 @@
           "approved": "KYC Approved!",
           "closed": "KYC Rejected"
         }
-      }
+      },
+      "filter": {
+        "all": "All",
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-desktop/static/i18n/en/app.json:432]: Is this viewable as part of this PR or ignore until the DEX/filter PR?

Comment (sarneijim) [reply apps/ledger-live-desktop/static/i18n/en/app.json:432]: yes, I am applying here the changes too advance the translations

**Comment by ggilchrist-ledger** [comment]
Branch looks good to me. Tickets have been created to fix small, non-blocking styling issues later

**Comment by ggilchrist-ledger** [comment]
@sarneijim Please investigate the quotes refetching before the timer has expired: 

https://user-images.githubusercontent.com/87659842/190450924-6c93786f-d917-426d-857d-b2165ee2ef2e.mov

**Pull request merged**